src/transports/PostMessageIframeTransport.ts

```diff
@@ -5,10 +5,12 @@ class PostMessageIframeTransport extends Transport {
   public uri: string;
   public frame: undefined | null | Window;
   public postMessageID: string;
+  public origin: string;
 
-  constructor(uri: string) {
+  constructor(uri: string, origin?: string) {
     super();
     this.uri = uri;
+    this.origin = origin || new URL(uri).origin;
```

Comment:
This is awesome thanks for doing this. The only thing here is that we're going to opt for a breaking change here. 

We're going to rm the default, because defaulting to origin can be a hidden breaking change to people using this in prod. 

We'll do a maj. verson bump which will force people to explicitly update their clients to set what domain they'd like to use this against, when using postmessage transport. 

Once this lands the team will bump the versions of generator and any other ecosystem consumers.

Reply:
```suggestion
    this.origin = origin;
```

Reply:
Commit message for these changes should be in the following format for semantic release

fix: patch security hole for post message iframe transport. this change forces consumers to
explicitly set the transport's acceptable origins. Improving the security using the transport.

BREAKING CHANGE: transport now takes second argument and requires origin be set for filtering
post message request.